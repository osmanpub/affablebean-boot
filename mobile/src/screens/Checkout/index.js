import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import CheckoutForm from '../../components/CheckoutForm';
import Confirmation from '../../components/Confirmation';
import Header from '../../components/Header';
import {ScrollView} from 'react-native';

class Checkout extends Component {
  render() {
    const {cart, purchase, setScreen} = this.props;
    const {order} = purchase;
    const details = order.hasOwnProperty('customer') ? (
      <Confirmation order={order} />
    ) : (
      <CheckoutForm {...this.props} />
    );

    return (
      <ScrollView>
        <Header cart={cart} currentScreen="Checkout" setScreen={setScreen} />
        {details}
      </ScrollView>
    );
  }
}

const mapStateToProps = state => {
  const {cart, purchase} = state;

  return {
    cart,
    purchase,
  };
};

export const ConnectedCheckout = connect(mapStateToProps)(Checkout);

Checkout.propTypes = {
  cart: PropTypes.object,
  dispatch: PropTypes.func.isRequired,
  purchase: PropTypes.object.isRequired,
  setScreen: PropTypes.func.isRequired,
};
