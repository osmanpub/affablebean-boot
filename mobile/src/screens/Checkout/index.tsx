import React from 'react';
import {ScrollView} from 'react-native';
import {connect} from 'react-redux';
import CheckoutForm from '../../components/CheckoutForm';
import Confirmation from '../../components/Confirmation';
import Header from '../../components/Header';
import {Purchase} from '../../interfaces/purchase';
import {RootState} from '../../redux';

type Props = {
  purchase: Purchase;
  setScreen: Function;
};

function Checkout(props: Props) {
  const {purchase, setScreen} = props;
  const {order} = purchase;
  const details = order.hasOwnProperty('customer') ? (
    <Confirmation order={order} />
  ) : (
    <CheckoutForm {...props} />
  );

  return (
    <ScrollView>
      <Header currentScreen="Checkout" setScreen={setScreen} />
      {details}
    </ScrollView>
  );
}

const mapStateToProps = (state: RootState) => ({
  purchase: state.purchase,
});

export default connect(mapStateToProps)(Checkout);
