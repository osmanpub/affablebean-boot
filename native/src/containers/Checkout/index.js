import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CheckoutForm from "../../components/CheckoutForm";
import Confirmation from "../../components/Confirmation";
import Header from "../../components/Header";

class Checkout extends Component {
  render() {
    const { cart, match, purchase } = this.props;
    const { order } = purchase;
    const details = order.hasOwnProperty("customer") ? (
      <Confirmation order={order} />
    ) : (
      <CheckoutForm {...this.props} />
    );

    return (
      <div>
        <Header cart={cart} url={match.url} />
        {details}
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart, purchase } = state;

  return {
    cart,
    purchase
  };
};

export const ConnectedCheckout = connect(mapStateToProps)(Checkout);

Checkout.propTypes = {
  cart: PropTypes.object,
  dispatch: PropTypes.func.isRequired,
  purchase: PropTypes.object.isRequired,
  setScreen: PropTypes.func.isRequired
};
