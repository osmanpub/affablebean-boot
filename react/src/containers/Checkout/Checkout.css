/* checkout page styles */
#checkoutTable {
  width: 600x;
  padding: 10px;
  background-color: #f5eabe;
  /*float: left;*/
  border-radius: 4px;
}

#checkoutTable td {
  vertical-align: top;
  text-align: right;
  margin: 5px;
}

.inputField {
  width: 190px;
}

#infoBox {
  margin-left: 150px;
  /*padding: 10px 25px 0 460px*/
}

#priceBox {
  padding: 10px;
  border: 1px solid #c3e3e0;
  background-color: #c3e3e0;
  border-radius: 4px;
}

#priceBox td {
  text-align: right;
  padding: 5px;
}

.checkoutPriceColumn {
  width: 80px;
  font-weight: bold;
}

.total {
  border-top: 1px solid #444;
}

.indent {
  margin-left: 6em;
}

/* confirmation page styles */
#confirmationText {
  margin: 0 20px 20px;
  padding: 10px;
  background-color: #f5eabe;
  float: left;
  border-radius: 4px;
}

.summaryColumn {
  margin-top: 15px;
  width: 50%;
  float: left;
}

.detailsTable {
  text-align: left;
}

.detailsTable th {
  padding: 5px 10px;
}

.detailsTable td {
  padding: 5px 10px 5px 20px;
  font-family: "Courier New", courier, monospace;
}

#orderSummaryTable {
  width: 100%;
}

#deliveryAddressTable {
  width: 70%;
  margin-left: 15%;
}

#deliveryAddressTable td {
  padding-top: 10px;
  padding-bottom: 10px;
}

.quantityColumn {
  text-align: center;
  width: 62px;
}

.confirmationPriceColumn {
  width: 90px;
  font-weight: normal;
}

#deliverySurchargeCellLeft {
  text-align: right;
  padding: 0;
}

#deliverySurchargeCellRight {
  padding: 0 20px;
}

#totalCellLeft {
  text-align: right;
  padding: 0;
}

#totalCellRight {
  padding: 0 20px;
}

#dateProcessedRow {
  padding: 2px 0 10px 20px;
}
