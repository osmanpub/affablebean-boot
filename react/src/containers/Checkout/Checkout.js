import React, { Component } from "react";
import { connect } from "react-redux";
import Categories from "../../components/Categories";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { fetchCategoriesIfNeeded } from "../../net/categories";

export class Checkout extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchCategoriesIfNeeded());
  }

  render() {
    const { cart, categories } = this.props;
    const { items } = categories;

    if (items.length === 0) {
      return null;
    }

    return (
      <div>
        <Header cart={cart} />
        <Categories categories={items} />
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { cart, categories } = state;

  return {
    cart,
    categories
  };
};

export default connect(mapStateToProps)(Checkout);
