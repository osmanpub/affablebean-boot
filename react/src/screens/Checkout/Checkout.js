import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CheckoutForm from "../../components/CheckoutForm";
import Confirmation from "../../components/Confirmation";
import Footer from "../../components/Footer";
import Header from "../../components/Header";

export class Checkout extends Component {
  render() {
    const { cart, match, purchase } = this.props;
    const { order } = purchase;
    const details = order.hasOwnProperty("customer") ? (
      <Confirmation order={order} />
    ) : (
      <CheckoutForm {...this.props} />
    );

    return (
      <div>
        <Header cart={cart} url={match.url} />
        {details}
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  cart: state.cart,
  purchase: state.purchase
});

export default connect(mapStateToProps)(Checkout);

Checkout.propTypes = {
  cart: PropTypes.object,
  match: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  purchase: PropTypes.object.isRequired
};
