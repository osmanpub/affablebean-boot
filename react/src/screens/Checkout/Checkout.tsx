import React from "react";
import { connect } from "react-redux";
import CheckoutForm from "../../components/CheckoutForm";
import Confirmation from "../../components/Confirmation";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Cart } from "../../interfaces/cart";
import { Purchase } from "../../interfaces/purchase";
import { RootState } from "../../redux";

type Props = {
  cart: Cart;
  match: any;
  purchase: Purchase;
};

export function Checkout(props: Props) {
  const { cart, match, purchase } = props;
  const { order } = purchase;
  const details = order.hasOwnProperty("customer") ? (
    <Confirmation order={order} />
  ) : (
    <CheckoutForm {...props} />
  );

  return (
    <div>
      <Header url={match.url} />
      {details}
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
  purchase: state.purchase
});

export default connect(mapStateToProps)(Checkout);
