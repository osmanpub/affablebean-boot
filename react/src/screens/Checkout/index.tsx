import React from "react";
import { connect } from "react-redux";
import CheckoutForm from "../../components/CheckoutForm";
import Confirmation from "../../components/Confirmation";
import Footer from "../../components/Footer";
import Header from "../../components/Header";
import { Purchase } from "../../interfaces/purchase";
import { RootState } from "../../redux";

type Props = {
  match: any;
  purchase: Purchase;
};

function Checkout(props: Props) {
  const { match, purchase } = props;
  const { order } = purchase;
  const details = order.hasOwnProperty("customer") ? (
    <Confirmation order={order} />
  ) : (
    <CheckoutForm {...props} />
  );

  return (
    <div>
      <Header url={match.url} />
      {details}
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  purchase: state.purchase
});

export default connect(mapStateToProps)(Checkout);
