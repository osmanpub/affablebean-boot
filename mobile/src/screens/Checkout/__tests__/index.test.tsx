import {act, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Checkout from '..';

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        product: {
          id: 1,
          description: 'semi skimmed (1L)',
          name: 'milk',
          price: 1.7,
        },
        quantity: 1,
        total: 1.7,
      },
    ],
    numberOfItems: 1,
    subtotal: 1.7,
  },
  purchase: {
    didInvalidate: false,
    isPosting: false,
    order: {},
  },
  ui: {
    formErrors: [],
  },
};

const store = mockStore(initialState);

const checkout = (
  <Provider store={store}>
    <Checkout setScreen={jest.fn()} />
  </Provider>
);

describe('<Checkout />', () => {
  it('renders correctly', () => {
    const component = renderer.create(checkout).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('show logo', () => {
    const {getByTestId} = render(checkout);
    const logo = getByTestId(/logo/i);
    expect(logo).toBeTruthy();
  });

  it('show view cart link', () => {
    const {getByTestId} = render(checkout);
    const viewCart = getByTestId(/view-cart/i);
    expect(viewCart).toBeTruthy();
  });

  it('show clear cart link', () => {
    const {getByTestId} = render(checkout);
    const clearCart = getByTestId(/clear-cart/i);
    expect(clearCart).toBeTruthy();
  });

  it('show 1 cart items', () => {
    const {getByText} = render(checkout);
    const items = getByText(/1 items/i);
    expect(items).toBeTruthy();
  });
});

describe('<Data Entry />', () => {
  it('renders correctly', () => {
    const component = renderer.create(checkout).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('show intro', () => {
    const {getByTestId} = render(checkout);
    const intro = getByTestId(/checkout-intro/i);
    expect(intro).toBeTruthy();
  });

  it('show 5 input fields', () => {
    const {getAllByTestId} = render(checkout);
    const inputs = getAllByTestId('checkout-input');
    expect(inputs).toHaveLength(5);
  });

  it('show submit button', () => {
    const {getByTestId} = render(checkout);
    const button = getByTestId(/checkout-submit/i);
    expect(button).toBeTruthy();
  });

  it("set name to 'osman'", () => {
    const {getByPlaceholderText} = render(checkout);
    const input = getByPlaceholderText('Enter your name');

    act(() => {
      fireEvent.change(input, {nativeEvent: {text: 'osman'}});
    });

    // @ts-ignore
    expect(input.props.value).toBe('osman');
  });

  it("set phone to '1234567890'", () => {
    const {getByPlaceholderText} = render(checkout);
    const input = getByPlaceholderText('Enter your phone number');

    act(() => {
      fireEvent.change(input, {nativeEvent: {text: '1234567890'}});
    });

    // @ts-ignore
    expect(input.props.value).toBe('1234567890');
  });

  it("set email to 'osman@gmail.com'", () => {
    const {getByPlaceholderText} = render(checkout);
    const input = getByPlaceholderText('Enter your email address');

    act(() => {
      fireEvent.change(input, {nativeEvent: {text: 'osman@gmail.com'}});
    });

    // @ts-ignore
    expect(input.props.value).toBe('osman@gmail.com');
  });

  it("set address to 'nobody cares where you live'", () => {
    const {getByPlaceholderText} = render(checkout);
    const input = getByPlaceholderText('Enter your address');

    act(() => {
      fireEvent.change(input, {
        nativeEvent: {text: 'nobody cares where you live'},
      });
    });

    // @ts-ignore
    expect(input.props.value).toBe('nobody cares where you live');
  });

  it("set cc to '1234567890123456'", () => {
    const {getByPlaceholderText} = render(checkout);
    const input = getByPlaceholderText('Enter your credit card number');

    act(() => {
      fireEvent.change(input, {nativeEvent: {text: '1234567890123456'}});
    });

    // @ts-ignore
    expect(input.props.value).toBe('1234567890123456');
  });
});
