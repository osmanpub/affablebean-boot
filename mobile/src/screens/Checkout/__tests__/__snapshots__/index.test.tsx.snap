// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> renders correctly 1`] = `
<ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/components/Header/logo.jpg",
          }
        }
        style={
          Object {
            "margin": 16,
          }
        }
      />
    </TouchableWithoutFeedback>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#f7f7e9",
          "padding": 16,
        }
      }
    >
      <Text>
        In order to purchase the items in your shopping cart, please provide us with the following information:
      </Text>
      <TextInput
        allowFontScaling={true}
        name="name"
        onChange={[Function]}
        placeholder="Enter your name"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        name="email"
        onChange={[Function]}
        placeholder="Enter your email address"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        name="phone"
        onChange={[Function]}
        placeholder="Enter your phone number"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        name="address"
        onChange={[Function]}
        placeholder="Enter your address"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        name="creditCard"
        onChange={[Function]}
        placeholder="Enter your credit card number"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "paddingTop": 16,
          }
        }
      >
        Subtotal: € 
        0.00
      </Text>
      <Text>
        Surcharge: € 
        3.00
      </Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Total: € 
        3.00
         
      </Text>
      <Text
        style={
          Object {
            "marginTop": 8,
          }
        }
      />
      <Button
        onPress={[Function]}
        title="Purchase"
      />
    </View>
  </View>
</ScrollView>
`;
